### PR TITLE
Possible infinite loop?

### PR DESCRIPTION
**Pull request by shivupa**

### CODE REVIEW

File: cclib/method/ddec.py [hunk 1]
```diff
@@ -736,6 +740,8 @@ def _converge_phi(self, phiA, superscript, atomi):
             candidates_phi.append(phiA)
             candidates_bigphi.append(bigphiA)
 
+        if curr_it == it_max:
```
Comment (langner) [inline cclib/method/ddec.py:743]: If you do this here, you don't need to tweak the loop definition.

**Comment by shivupa** [comment]
😃

File: test/data/skip.py [hunk 2]
```diff
@@ -31,3 +33,27 @@ def testwrapper(self, *args, **kwargs):
                 testfunc(self, *args, **kwargs)
         return testwrapper
     return testdecorator
+
+def skipForVersion(version, msg):
```
Comment (amandadumi) [inline test/data/skip.py:37]: the pytest built in function may just let us have less code to maintain? https://docs.pytest.org/en/7.1.x/how-to/skipping.html

For our purposes:
`@pytest.mark.skipif(sys.version_info < (3, 8), reason="his test doesn't converge with newer psi4 versions >3.8")`

**Review by amandadumi — CHANGES_REQUESTED** [review]
Other than one suggestion, this looks good!

File: test/data/skip.py (hunk 2 above)
Comment (shivupa) [reply test/data/skip.py]: Fixed!

**Comment by shivupa** [comment]
Ready for review

**Pull request merged**